### PR TITLE
Add: Block body validation.

### PR DESCRIPTION
**Pull request by nc6**
TODO: Split aside the update rules, handle the case where there is no update
proposal.

### CODE REVIEW

File: specs/chain/latex/blockchain-spec.tex [hunk 1]
```diff
@@ -772,6 +780,10 @@ \subsection{Block body processing}
   \begin{equation*}
     \inference
     { \maxblocksize \mapsto \var{b_{max}} \in \var{pps} && \bsize{b} \leq \var{b_{max}} \\
+      \var{bh} \leteq \bhead{b} & \var{vk_d} \leteq \bissuer{\var{bh}} \\
```
Comment (dnadales) [inline specs/chain/latex/blockchain-spec.tex:783]: Now that we have a bunch of verification here we might want to factor that out into a separate rule.

File: specs/chain/latex/blockchain-spec.tex [hunk 2]
```diff
@@ -717,9 +720,12 @@ \subsection{Block body processing}
       \fun{bUtxo} & \Block \totalf \UTxO & \text{Block UTxO payload} \\
       \fun{\bcertsname} & \Block \totalf \seqof{\DCert}
       & \text{block certificates} \\
-      \fun{bUpdProp} & \Block \totalf \UProp & \text{Block update proposal payload} \\
+      \fun{bUpdProp} & \Block \totalf \UProp^{?} & \text{Block update proposal payload} \\
```
Comment (dnadales) [inline specs/chain/latex/blockchain-spec.tex:723]: We need to update the UPIREG rule to work with this kind of maybe type.

**Review by dnadales — APPROVED** [review]

File: specs/chain/latex/blockchain-spec.tex (hunk 2 above)
Comment (nc6) [reply specs/chain/latex/blockchain-spec.tex:723]: Well, either that or have a wrapper. I was thinking of factoring the three update rules out and having that taken care of there.

**Pull request merged**